doc/user/outputs/database.rst

```diff
@@ -8,6 +8,52 @@ the text of the input files that were used to create the case, and for each time
 the values of all composite parameters as well as layout information to help fully
 reconstruct the structure of the reactor model.
 
+Loading Reactor State
+=====================
+Among other things, the database file can be used to recover an ARMI reactor model from
+any of the time nodes that it contains. This can be useful for performing restart runs,
+or for doing custom post-processing tasks. To load a reactor state, you will need to
+open the database file into a ``Database`` object. From there, you can call the
+:py:meth:`armi.bookkeeping.db.Database3.load()` method to get a recovered
+reactor object. For instance, given a database file called ``myDatabase.h5``, we could
+load the reactor state at cycle 5, time node 2 with the following::
+
+   from armi.bookkeeping.db import databaseFactory
+
+   db = databaseFactory("myDatabase.h5", "r")
```

Comment:
It might be worthwhile at some point to have a default of "r" to the permission argument (not in this diff).  The vast majority of time people will call this in scripts, it will be "r". To get a reactor, needing databasePath, cycle, and timeNode is pretty elegant. Needed to remember "r" as well is less so.

Reply:
Optional arguments lead to ambiguity, which normally requires looking up the function docstring to resolve, so it would only be elegant to a person that is already intimately familiar with the API. Also, having the second argument lends parity all of the built-in `open()`-like functions throughout the python ecosystem.

Reply:
open() has a default of mode 'r' (just like I am suggesting)
https://docs.python.org/3/library/functions.html#open

I think making it a kwarg would lend more parity. Note also that open() had quite a few other kwargs that normal people don't use on a daily basis, but this doesn't mean I have to look up the doc strings constantly to resolve. I can't think of a time that I ever had a armi script where I want to load in the db in 'w' mode, but 'r' mode is extremely common.